### DISCUSSION TITLE
Trying to migrate to 4.38 but can not get rid of jwt_secret warning

### DISCUSSION THREAD

**Discussion by Nutellaeis** [authelia/authelia#6954] [Troubleshooting] [answered]
As I am currently trying to migrate to 4.38 I am trying to get rid of all the warnings. The last warning I can not get rid of is:

`time="2024-03-19T09:35:19Z" level=warning msg="Configuration: configuration key 'jwt_secret' is deprecated in 4.38.0 and has been replaced by 'identity_validation.reset_password.jwt_secret': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"`

I do not even have an `identity_validation` section in my configuration, as my jwt secret is defined in docker compose as:
`AUTHELIA_JWT_SECRET_FILE: /secrets/JWT_SECRET` As the other values are not required they are not set. Should the variable not be recognized as a new configuration?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

To resolve the warning you're encountering after migrating to Authelia version 4.38, you need to adjust your configuration according to the new standards introduced in this version. The warning message indicates that the `jwt_secret` configuration key is deprecated and has been replaced by `identity_validation.reset_password.jwt_secret`. 

Since you're using Docker Compose and have your JWT secret defined as an environment variable (`AUTHELIA_JWT_SECRET_FILE`), you should now use the new configuration key in your environment variable. However, based on the information provided in the context, there isn't a direct mention of changing the environment variable names to reflect the new configuration paths.

Given the structure of Authelia's configuration and the warning message, it seems like your current setup should technically work as Authelia should automatically map the deprecated key to the new one. However, to explicitly follow the new configuration structure and avoid the warning, you might consider adjusting your configuration file (if you have one) to include the `identity_validation` section with the `reset_password` subsection, specifying the `jwt_secret` there.

Since you mentioned not having an `identity_validation` section and relying on Docker Compose for setting the JWT secret, it's possible that Authelia expects the new configuration path to be reflected in the environment variable as well. Unfortunately, the provided context does not explicitly state a change in environment variable naming conventions to match the new configuration paths.

To proceed without further warnings and ensure future compatibility (especially with the upcoming version 5.0.0 where the auto-mapping might be removed), I recommend checking Authelia's official documentation or GitHub repository for any updates or examples that clarify how to set the JWT secret using environment variables according to the new configuration structure. If no clear guidance is available, consider reaching out to the Authelia support or community forums for clarification.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
New env name is `AUTHELIA_IDENTITY_VALIDATION_RESET_PASSWORD_JWT_SECRET_FILE` as an optional configuration update (i.e. deprecated and remapped for you).

**Reply by Nutellaeis** [reply]
That makes perfect sense. Thanks for the quick reply. I must have missed that change in the docs.

**Reply by james-d-elliott** [reply]
Well we didn't expressly list them all because realistically people can ignore the warnings for now and do it as they see fit. It is likely one we didn't expressly mention because it was probably more straight forward than the others, but we may have.

Secrets information is here: https://www.authelia.com/configuration/methods/secrets/
